internal/app/service.go

```diff
@@ -118,12 +118,25 @@ type contextGroup struct {
 
 // AddRoute allows you to leverage the existing webserver to add routes.
 func (svc *Service) AddRoute(route string, handler func(nethttp.ResponseWriter, *nethttp.Request), methods ...string) error {
+	// Legacy behavior is to add unauthenticated route
+	return svc.AddCustomRoute(route, interfaces.Unauthenticated, handler, methods...)
+}
+
+// AddCustomRoute allows you to leverage the existing webserver to add routes.
+func (svc *Service) AddCustomRoute(route string, authentication interfaces.Authentication, handler func(nethttp.ResponseWriter, *nethttp.Request), methods ...string) error {
```

Comment:
This 3.1 enhancement can be removed and just focus on securing AddRoute as it was done originally in Minnesota, like the Device SDK was.